<launch>

    <arg name="camera" default="xtion"/>
    <arg name="ranging_radio" default="true"/>
    <arg name="radio_id"/>
    <arg name="tf_prefix"/>

    <group ns="$(arg tf_prefix)">

        <!-- Launch robot components -->
        <include file="$(find robot_io)/launch/robot.launch">
            <arg name="camera"          value="$(arg camera)"/>
            <arg name="hokuyo"          value="true"/>
            <arg name="odometry"        value="true"/>
            <arg name="tf_prefix"       value="$(arg tf_prefix)"/>
            <arg name="pulson"          value="$(arg ranging_radio)"/>
            <arg name="radio_id"        value="$(arg radio_id)"/>
        </include>

        <!-- panorama nodelet -->
        <include file="$(find panorama)/launch/panorama_nodelet.launch">
          <arg name="save_directory"      value="/home/odroid/.ros/panoramas"/>
          <arg name="world_frame"         value="$(arg tf_prefix)/map"/>
          <arg name="camera_frame"        value="$(arg tf_prefix)/$(arg camera)_rgb_frame"/>
          <arg name="robot_frame"         value="$(arg tf_prefix)/base_link"/>
          <arg name="odom_frame"          value="$(arg tf_prefix)/odom"/>
          <arg name="image_registration"  value="true"/>
          <arg name="debug"               value="false"/>
          <arg name="spin_speed"          value="0.25"/>
        </include>

        <!-- camera transform tree -->
        <include file="$(find robot_io)/launch/xtion_frames.launch">
            <arg name="camera"    value="$(arg camera)" />
            <arg name="tf_prefix" value="$(arg tf_prefix)" />
        </include>

        <!-- onboard localization provided gmapping -->
        <include file="$(find robot_io)/launch/gmapping.launch">
            <arg name="camera"      value="$(arg camera)" />
            <arg name="tf_prefix"   value="$(arg tf_prefix)" />
        </include>

        <!-- exploration manager -->
        <include file="$(find csqmi_exploration)/launch/robot.launch">
          <arg name="tf_prefix" value="$(arg tf_prefix)"/>
        </include>

    </group>

</launch>
